### PR TITLE
Merge: complete rewrite

### PR DESCRIPTION
**Pull request by le-jeu**
This PR splits the merge feature from the operation model.
The merge feature follow a rebase mechanism using the common ancestor of the operations to be merged.
We always have a `master` copy (server), the `follower` (local) and the ancestor `origin` that was saved in `local` data at last fetch.
The merge is structured in 4 steps:
 1. `masterChange`: difference between `master` and `origin` (what's new on the server)
     `followerChange`: difference between `follower` and `origin` (what did I changed or fb event changed)
 2. `followerChange - masterChange`: compute what are the local changes that are not on `master`
     conflict may arise: we change smth that is not on `master` anymore, we change something both on `master` and `follower` etc
 3. if there are conflicts (that we don't want to solve automatically), the user needs to choose for every conflicts between the `master` and `follower` version
 4. apply on `master` what was previously computed and uses the resolved conflicts at step 3

The merge dialog is reworked to match the step 3.

`operation.changes()` is simplified to return a boolean instead of the computed changes.

### CODE REVIEW

File: src/code/model/changes.ts [hunk 1]
```diff
@@ -0,0 +1,609 @@
...
+/**
+ * @returns summary of marker differences, including the marker ID
+ */
+function markerChanges(origin: WasabeeMarker, current: WasabeeMarker) {
+  const changes: MarkerChange = {
+    id: origin.ID,
+    type: "edition",
+    props: computeDiff(origin, current, [
+      "type",
+      /*"portalId",*/ // unlikely because we don't swap marker yet
```
Comment (cloudkucooland) [inline src/code/model/changes.ts:102]: I thought we'd added (or were planning on adding) maker swap

File: src/code/model/operation.ts [hunk 2]
```diff
@@ -1259,88 +1260,81 @@ export default class WasabeeOp extends Evented implements IOperation {
...
-      )
-        this.localchanged = false;
+      this.localchanged = this.changes();
     }
     return this.localchanged;
   }
 
   mergeZones(op: WasabeeOp) {
-    const ids = new Set();
+    const ids = new Map<ZoneID, WasabeeZone>();
```
Comment (cloudkucooland) [inline src/code/model/operation.ts:1334]: yes!

**Review by cloudkucooland — COMMENTED** [review]
From what I'm able to follow, this looks good. I'm still slow at reading typescript and I don't know these codepaths well, but nothing jumped out at me as being wrong.

File: src/code/model/changes.ts (hunk 1 above)
Comment (le-jeu) [reply src/code/model/changes.ts:102]: "yet", also what swaping marker does has to be defined first, 
preserving the link ID on anchor swap allow to track anchor and resolve situation where one moved an anchor while another assigned it
imo, marker don't need this refinement and can change ID on swap

**Pull request merged**